frontend/src/components/nav/navbar.js

```diff
@@ -112,9 +122,12 @@ class NavBar extends React.Component {
                             onKeyUp={this.getSearchInput}   
                             value={this.state.word}
                             onChange={this.update()}
-                            onKeyPress={this.searchFunc()}
+                            // onKeyPress={this.searchFunc()}
+                            onKeyPress={this.debounceSearch(this.searchFunc(), 250)}
```

Comment:
this.searchFunc() invokes the function. To pass this.searchFunc as an argument, you need to do this instead:

{this.debounceSearch(this.searchFunc, 250)}